llvm/lib/Target/AIE/AIETargetMachine.cpp

```diff
@@ -38,11 +39,32 @@
 #include "llvm/InitializePasses.h"
 #include "llvm/MC/TargetRegistry.h"
 #include "llvm/PassRegistry.h"
+#include "llvm/Passes/PassBuilder.h"
 #include "llvm/Support/CodeGen.h"
 #include "llvm/Support/FormattedStream.h"
 #include "llvm/Target/TargetOptions.h"
+#include "llvm/Transforms/IPO/GlobalDCE.h"
+#include "llvm/Transforms/IPO/Internalize.h"
 using namespace llvm;
 
+static cl::opt<bool>
+    EnableCustomAliasAnalysis("aie-enable-alias-analysis",
+                              cl::desc("Enable AIE alias analysis pass"),
+                              cl::init(true), cl::Hidden);
+
+// Option to run internalize pass.
+static cl::opt<bool> InternalizeSymbols(
+    "aie-internalize-symbols",
+    cl::desc("Enable elimination of non-kernel functions and unused globals"),
+    cl::init(false), cl::Hidden);
```

Comment:
As discussed offline, I think this should be enabled by default.

Reply:
Discussed offline, static libraries can be a problem. We should only internalize and simplify if we are producing an "final executable". Let's keep it disabled by default.